appsec/src/extension/commands_helpers.c

```diff
@@ -553,6 +559,41 @@ static void _set_appsec_span_data(mpack_node_t node)
     }
 }
 
+void dd_command_process_settings(mpack_node_t root)
+{
+    if (mpack_node_type(root) != mpack_type_map) {
+        return;
+    }
+
+    size_t count = mpack_node_map_count(root);
+
+    for (size_t i = 0; i < count; i++) {
+        mpack_node_t key = mpack_node_map_key_at(root, i);
+        mpack_node_t value = mpack_node_map_value_at(root, i);
+
+        if (mpack_node_type(key) != mpack_type_str) {
+            mlog(dd_log_warning,
+                "Failed to process setting: invalid type for key");
+            return;
+        }
+        if (mpack_node_type(value) != mpack_type_str) {
+            mlog(dd_log_warning,
+                "Failed to process setting: invalid type for value");
+            return;
+        }
+
+        const char *key_str = mpack_node_str(key);
+        const char *value_str = mpack_node_str(value);
+        size_t key_len = mpack_node_strlen(key);
+        size_t value_len = mpack_node_strlen(value);
+
+        if (dd_string_equals_lc(
+                key_str, key_len, ZEND_STRL("auto_user_instrum"))) {
+            dd_parse_user_collection_mode_rc(value_str, value_len);
```

Comment:
Neither function `dd_parse_user_collection_mode_rc` or here there is logging when something unexpected happen. Could you please add some?